transports/tls/src/upgrade.rs

```diff
@@ -57,6 +65,17 @@ impl Config {
             client: crate::make_client_config(identity, None)?,
         })
     }
+
+    pub(crate) fn with_remote_peer_id(
+        remote_peer_id: Option<PeerId>,
+    ) -> Result<Self, certificate::GenError> {
+        let identity = libp2p_identity::Keypair::generate_ed25519();
+
+        Ok(Self {
+            server: crate::make_server_config(&identity)?,
+            client: crate::make_client_config(&identity, remote_peer_id)?,
+        })
+    }
```

Comment:
Instead of having this, I'd prefer if we instead store the `identity::Keypair` within `Config` and call the `make_server_config` and `make_client_config` as part of the upgrade.